src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
Yeah.  I saw the diffs in a few places in framework assemblies during ngen.  I have been unable to get a reduced repro to show the same behavior unfortunately.  But the tree looked like:

               [000052] ------------             *  stmtExpr  void  (top level) (IL   ???...  ???)
               [000049] ---XG-------             |  /--*  cast      int <- ubyte <- int
               [000047] ------------             |  |  |  /--*  const     int    64
               [000048] ---XG-------             |  |  \--*  |         int   
               [000046] *--XG-------             |  |     \--*  indir     ubyte 
               [000044] ------------             |  |        \--*  lclVar    byref  V05 tmp3         
               [000051] -A-XG-------             \--*  =         byte  
               [000050] *------N----                \--*  indir     byte  
               [000045] ------------                   \--*  lclVar    byref  V05 tmp3